src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -253,18 +256,39 @@ private void removeCGFromMonitors(CertificateGenerator gen) {
         cisShadowMonitor.removeFromMonitor(gen);
     }
 
+    /**
+     * Configures the KeyStore to use a certificates provided from the CA configuration.
+     *
+     * @throws KeyStoreException if the stored key is not a private key
+     * @throws Exception if failing to get the keyPair or certificateChain
+     * @throws IllegalArgumentException if configuration is invalid for CA
+     */
+    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
+    public void configureCustomCA() throws KeyStoreException, Exception {
+        if (!caConfiguration.isUsingCustomCA()) {
+            String errMsg = "Invalid configuration: certificateUri and privateKeyUri are required.";
```

Comment:
create a static final string instead of an inline string. Or just put it in the exception constructor without an intermediate variable.

Reply:
Yeah this was more statistical I didn't like how it looked putting it directly in the constructor but I can change that